lib/mayaUsd/CMakeLists.txt

```diff
@@ -49,6 +49,9 @@ target_compile_definitions(${PROJECT_NAME}
         $<$<BOOL:${IS_LINUX}>:LINUX>
         $<$<BOOL:${IS_LINUX}>:GL_GLEXT_PROTOTYPES>
         $<$<BOOL:${IS_LINUX}>:GLX_GLXEXT_PROTOTYPES>
+
+        # this flag is needed when building maya-usd in Maya
+        $<$<BOOL:${IS_WINDOWS}>:WIN32>
```

Comment:
Without 'WIN32' we get the following compiler error:
```
maya\src\components\usd\maya-usd\lib\mayausd\ufe\ProxyShapeHierarchyHandler.h(39): warning C4275: non dll-interface class 'Ufe_v0::HierarchyHandler' used as base for dll-interface class 'MayaUsd_v0::ufe::ProxyShapeHierarchyHandler'
 ufe-0.2.12\common\include\ufe/hierarchyHandler.h(25): note: see declaration of 'Ufe_v0::HierarchyHandler'
```
This flags was previously set in MayaUSD.cmake via CMAKE_CXX_FLAGS. CMAKE_CXX_FLAGS from other cmake echo systems no longer influence maya-usd.  

There were some compiler flags/defs (e.g _WINDOWS, /GR) which I don't see why is needed.

Reply:
The reason I added those three flags in our internal MayaUSD.cmake was to fix some linker errors when compiling the repo from inside our internal Maya build. Those three flags came from comparing the standalone compilation vs the one inside of Maya. At the time I remember searching the maya-usd repo to find where these three flags were added, but I could not find them.